src/Microsoft.Data.SqlClient/netcore/ref/Microsoft.Data.SqlClient.cs

```diff
@@ -1443,9 +1443,13 @@ public sealed partial class SqlException : System.Data.Common.DbException
         public override string Source { get { throw null; } }
         /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlException.xml' path='docs/members[@name="SqlException"]/State/*'/>
         public byte State { get { throw null; } }
+
         /// <include file='../../../../doc/snippets/Microsoft.Data.SqlClient/SqlException.xml' path='docs/members[@name="SqlException"]/GetObjectData/*'/>
 #if !NET6_0_OR_GREATER
         [System.Security.Permissions.SecurityPermissionAttribute(System.Security.Permissions.SecurityAction.LinkDemand, Flags = System.Security.Permissions.SecurityPermissionFlag.SerializationFormatter)]
+#endif
+#if NET8_0_OR_GREATER
+        [System.Obsolete]
```

Comment:
Based on [.net doc](https://learn.microsoft.com/en-us/dotnet/api/system.exception.getobjectdata?view=net-8.0&devlangs=csharp&f1url=%3FappId%3DDev16IDEF1%26l%3DEN-US%26k%3Dk(System.Exception.GetObjectData)%3Bk(SolutionItemsProject)%3Bk(DevLang-csharp)%26rd%3Dtrue) suggestion.

Should this be applied on API docs as well?
cc @David-Engel

```suggestion
        [Obsolete("This API supports obsolete formatter-based serialization. It should not be called or extended by application code.", DiagnosticId="SYSLIB0051", UrlFormat="https://aka.ms/dotnet-warnings/{0}")]
```